src/pam_hbac_obj.c

```diff
@@ -172,6 +172,27 @@ get_user_groups(const char *name, gid_t primary_gid,
         ret = 0;
         *ngroups_ptr = ngroups;
     }
+#elif defined(HAVE_GETGRSET)
+    int ngroups;
+    long max_group_len;
+    char *gid_list_s, *gid_s;
+
+    ngroups = 0;
+    max_group_len = sysconf(_SC_LOGIN_NAME_MAX);
+
+    /* string containing comma separated list of gids the user belongs to */
+    gid_list_s = malloc(sizeof(char)*max_group_len*NGROUPS_MAX);
```

Comment:
Hmm, here are two lines that both assign to gid_list_s and both allocate memory, so the memory that comes from malloc is lost.. At least some documentation I found (https://www.ibm.com/support/knowledgecenter/ssw_aix_71/com.ibm.aix.basetrf1/getgrset.htm) seems to suggest that `getgrset` already returns an allocated list, therefore I think the malloc andthe max_group_len variable are not needed?

Reply:
I think you are right, no need for the malloc and the variable. My C programming skills are not the best :).